core/src/main/java/com/linecorp/armeria/client/DefaultHttpClient.java

```diff
@@ -42,22 +45,38 @@ public HttpResponse execute(HttpRequest req) {
     }
 
     private HttpResponse execute(@Nullable EventLoop eventLoop, HttpRequest req) {
-        final String originalPath = req.path();
-        final String newPath = concatPaths(uri().getRawPath(), originalPath);
-        final PathAndQuery pathAndQuery = PathAndQuery.parse(newPath);
-        if (pathAndQuery == null) {
-            req.abort();
-            return HttpResponse.ofFailure(new IllegalArgumentException("invalid path: " + newPath));
-        }
-
-        final HttpRequest newReq;
-        if (newPath != originalPath) {
-            newReq = HttpRequest.of(req, req.headers().toBuilder().path(newPath).build());
+        if (uri() == UNDEFINED_URI) {
+            final URI uri = URI.create(req.path());
+            final String authority = uri.getAuthority();
+            if (authority == null) {
+                return HttpResponse.ofFailure(new IllegalArgumentException("no authority: " + req.path()));
+            }
+            final Endpoint endpoint = Endpoint.parse(authority);
```

Comment:
Do we need to cache `Endpoint`?

Reply:
I think it's ok without caching, absolute URLs still probably aren't such a common request, and it's just one small object.

Technically, `before` is creating a whole new `HttpClient` to handle absolute URLs, so it's actually much less GC than that :P

Reply:
I agreed with `Endpoint` is a small object. I kept the current logic. :-)